tests/integration/release/high_availability/test_upgrade_from_stable.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
q: I see parts of this code are very similar to `tests/integration/integration/h_a/test_upgrade_from_stable.py`, is there a plan to create some common helpers?